app/services/fees/charge_service.rb

```diff
@@ -37,7 +38,12 @@ def call
 
       ActiveRecord::Base.transaction do
         result.fees.each do |fee|
```

Comment:
Perhaps it's more expressive to do `result.fees.filter! {|f| should_persist_fee(f)}` before this loop? 

(specifically to avoid confusion around `next result.fees.delete(fee)`)